core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -426,7 +426,6 @@ private boolean isColumnsEnumeratingTable(InformationSchemaTable table)
         if (domain.getValues() instanceof SortedRangeSet) {
             ImmutableSet.Builder<String> result = ImmutableSet.builder();
             for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
-                checkState(!range.isAll()); // Already checked
```

Comment:
so this assertion was not correct? why?

Reply:
why would it be correct?

Reply:
why it was added in first place?

Reply:
why it was added in first place?

Reply:
that's a question to @sopel39 (see https://github.com/trinodb/trino/commit/625712ba24fd547180326ea754394dd9a660ebb4) :)

Reply:
Ah, I see. I copied that section from somwhere. If you grep though code `checkState(!range.isAll()); // Already checked` you will find many similar places. I wonder if all of them are correct (some look correct)

Reply:
i checked all of them and the only invalid one is covered by https://github.com/trinodb/trino/issues/9885